### DISCUSSION TITLE
Ollama JSON mode

### DISCUSSION THREAD

**Discussion by pixelsquared** [datacrystals/AIStoryWriter#19] [General]
Have you considered using Ollama's JSON mode for the JSON generation? https://github.com/ollama/ollama/blob/main/docs/api.md#json-mode

**Comment by datacrystals** [comment]
I had not - that would be vastly preferable to how it's being done right now. Thanks! 

Do you have time to replace the existing JSON system with the better Ollama method and open a PR?

**Reply by datacrystals** [reply]
@LoggeL has implemented this, and merged it! Please feel free to let us know what you think.

**Comment by LoggeL** [comment]
> Have you considered using Ollama's JSON mode for the JSON generation? https://github.com/ollama/ollama/blob/main/docs/api.md#json-mode

Thanks for the input. It definitely reduced the json generation errors. Sadly the LLMs still manage to outsmart the JSON parser by escaping etc which sometimes still causes errors but now it's faster :)
